Do not report warning event when an unknown provisioner is requested.

jsafrane opened the pull request:
with `StorageClass.Provisioner == <unknown plugin>`, we should wait for
either external provisioner or volume admin to provide a PV for a claim
instead of reporting an error.

Fixes #31723

<!-- Reviewable:start -->

---

This change is [<img src="https://reviewable.kubernetes.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/32077)

<!-- Reviewable:end -->

thockin commented:
LGTM, can wait until after 1.4.0, though, maybe even to 1.5 ?

thockin commented on pkg/controller/volume/persistentvolume/controller.go:
```diff
@@ -1234,6 +1235,19 @@ func (ctrl *PersistentVolumeController) provisionClaimOperation(claimObj interfa
+		// and wait for the external provisioner
+		if storageClass != nil {
+			msg := fmt.Sprintf("cannot find provisioner %q, expecting that a volume for the claim is provisioned either manually or via external software", storageClass.Provisioner)
+			ctrl.eventRecorder.Event(claim, api.EventTypeNormal, "ExternalProvisioning", msg)
```
I don't know that this warrants an event, but I am OK with it.

jsafrane committed:
Do not report warning event when an nknown provisioner is requested

with StorageClass.Provisioner == <unknown plugin>, we should wait for
either external provisioner or volume admin to provide a PV for a claim
instead of reporting an error.

Fixes #31723

jsafrane commented:
rebased

jsafrane commented:
restoring lgtm, shifting to 1.5.

jeffvance commented:
What is the Phase of a PV that is waiting to be deleted? Can this PV be re-claimed? If the PV is not claimable then should the Phase reflect that it is waiting-to-be-deleted?
